Source/missiles.cpp

```diff
@@ -762,6 +763,92 @@ bool CanAddEffect(const PlayerStruct &player, missile_id type)
 	return true;
 }
 
+/**
+ * @brief Find parent missile (uses _miVar4)
+ *
+ * @return id of parent missile or -1
+ */
+int FindParent(MissileStruct &missile)
+{
+	for (int i = 0; i < ActiveMissileCount; i++) {
+		int mx = ActiveMissiles[i];
+		auto &parent = Missiles[mx];
+		if (parent._misource != missile._misource)
+			continue;
+		int childId = -(parent._miVar4 + 1);
+		if (childId < 0 || childId >= MAXMISSILES)
+			continue;
+		if (&Missiles[childId] != &missile)
+			continue;
+
+		return mx;
+	}
+
+	return -1;
+}
+
+/**
+ * @brief Sync missile position with parent missile, by searching for self reference in all mssiles
+ *
+ * @todo It's unclear if this is actually neede is mostly dead code
+ */
+void SyncPositionWithParent(MissileStruct &missile)
+{
+	int mx = FindParent(missile);
+	if (mx == -1)
+		return;
+
+	auto &parent = Missiles[mx];
+
+	missile.position.offset = parent.position.offset;
+	missile.position.traveled = parent.position.traveled;
+}
+
+void SpawnLightning(MissileStruct &missile, int dam)

```

Comment:
🎉

Reply:
i dont get it :)

Reply:
Glad to see a dedup of so many lines of code, that's all :D

Reply:
lol thought it was a 🍕 emoji 😂